packages/aws-cdk/bin/cdk.ts

```diff
@@ -382,6 +393,30 @@ async function initCommandLine() {
   }
 }
 
+async function lookupProjectRoot(): Promise<string | undefined> {
+
+  async function ascend(directoryParts: string[]): Promise<string | undefined> {
+
+    if (directoryParts.length === 0) {
+      return undefined;
+    }
+
+    const directory = directoryParts.join(path.sep);
+
+    const filePath = path.join(directory, PROJECT_CONFIG);
+
+    if (await fs.pathExists(filePath)) {
+      return directory;
+    }
+
+    return ascend(directoryParts.slice(0, -1));
+
+  }
+
+  return ascend(path.resolve(process.cwd()).split(path.sep));
+
+}
+
```

Comment:
I guess I'd just use `path.dirname` instead of parsing the path:

```ts
async function ascend(dir: string) {
  if (dir === path.sep) { return undefined; }
  // ...
  return ascend(path.dirname(dir));
}
```